File: 351004/Leshok/lab3/publisher/src/controllers/CreatorController.js
const CreatorService = require('../services/CreatorService');
const { CreatorRequestDto } = require('../dtos/CreatorDto');

class CreatorController {
  async create(req, res, next) {
    try {
      const dto = new CreatorRequestDto(req.body.login, req.body.password, req.body.firstname, req.body.lastname);
      const result = await CreatorService.create(dto);
      res.status(201).json(result);
    } catch (err) { next(err); }
  }

  async update(req, res, next) {
    try {
      const dto = new CreatorRequestDto(req.body.login, req.body.password, req.body.firstname, req.body.lastname);
      const result = await CreatorService.update(req.params.id, dto);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async getOne(req, res, next) {
    try {
      const result = await CreatorService.getById(req.params.id);
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async getAll(req, res, next) {
    try {
      const result = await CreatorService.getAll();
      res.status(200).json(result);
    } catch (err) { next(err); }
  }

  async delete(req, res, next) {
    try {
      await CreatorService.delete(req.params.id);
      res.status(204).send();
    } catch (err) { next(err); }
  }
}

module.exports = new CreatorController();

File: 351003/Kisel/JPA/RV_Kisel_lab2_Task320/Controllers/CreatorController.cs
using Microsoft.AspNetCore.Mvc;
using RV_Kisel_lab2_Task320.Models.Dtos;
using RV_Kisel_lab2_Task320.Services;

namespace RV_Kisel_lab2_Task320.Controllers;

[ApiController]
[Route("api/v1.0/creators")] // Явный путь для тестов
public class CreatorController : ControllerBase {
    private readonly ICreatorService _service;
    public CreatorController(ICreatorService service) => _service = service;

    [HttpGet] 
    public async Task<ActionResult> GetAll() => Ok(await _service.GetAllAsync());

    [HttpGet("{id}")] 
    public async Task<ActionResult> GetById(int id) {
        var r = await _service.GetByIdAsync(id);
        if (r == null) return NotFound(new ErrorResponse { ErrorMessage = "Not Found", ErrorCode = "40401" });
        return Ok(r);
    }

    [HttpPost] 
    public async Task<ActionResult> Create(CreatorDto dto) {
        var all = await _service.GetAllAsync();
        if (all.Any(x => x.Login == dto.Login)) 
            return StatusCode(403, new ErrorResponse { ErrorMessage = "Duplicate login", ErrorCode = "40301" });

        var result = await _service.CreateAsync(dto);
        return CreatedAtAction(nameof(GetById), new { id = result.Id }, result);
    }

    [HttpPut] // Метод для обновления (без ID в URL, как в логах теста)
    public async Task<ActionResult> Update(CreatorDto dto) {
        await _service.UpdateAsync(dto.Id, dto);
        return Ok(dto);
    }

    [HttpDelete("{id}")] 
    public async Task<IActionResult> Delete(int id) {
        await _service.DeleteAsync(id);
        return NoContent();
    }
}